proxy/src/graphql/schema.rs

```diff
@@ -366,6 +381,24 @@ impl ControlMutation {
 
         Ok(true)
     }
+
+    fn register_user(
+        ctx: &Context,
+        handle: juniper::ID,
+        id: juniper::ID,
+    ) -> Result<registry::Transaction, error::Error> {
+        // TODO(xla): Get keypair from persistent storage.
+        let fake_pair = ed25519::Pair::from_legacy_string("//Alice", None);
+        // TODO(xla): Use real fee define dby the user.
+        let fee = 100;
+
+        futures::executor::block_on(
+            ctx.registry
+                .read()
```

Comment:
I'm not fully sure as to what this is but should it be `write` since it's registering a user and not, say, getting/reading one?

Reply:
These are the calls to acquire the `RWLock` and as we don't mutate the field itself we  don't need to acquire a `write` lock.